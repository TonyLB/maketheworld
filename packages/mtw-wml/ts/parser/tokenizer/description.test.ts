import SourceStream from './sourceStream'
import descriptionTokenizer from './description'

describe('descriptionTokenizer', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.resetAllMocks()
    })

    it('should return undefined when no match', () => {
        const testStream = new SourceStream('<Room />')
        expect(descriptionTokenizer(testStream)).toBe(undefined)
    })
    it('should tokenize to the next whitespace', () => {
        const testStream = new SourceStream('Test Message</Room>')
        expect(descriptionTokenizer(testStream)).toMatchSnapshot()
    })
    it('should tokenize to the next tag', () => {
        const testStream = new SourceStream('Test</Room>')
        expect(descriptionTokenizer(testStream)).toMatchSnapshot()
    })
    it('should tokenize to end of source', () => {
        const testStream = new SourceStream('Test')
        expect(descriptionTokenizer(testStream)).toMatchSnapshot()
    })
})