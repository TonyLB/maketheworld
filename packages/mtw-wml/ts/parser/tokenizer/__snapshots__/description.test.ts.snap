// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`descriptionTokenizer should tokenize to end of source 1`] = `
Object {
  "endIdx": 4,
  "source": "Test",
  "startIdx": 0,
  "type": "Description",
}
`;

exports[`descriptionTokenizer should tokenize to the next tag 1`] = `
Object {
  "endIdx": 4,
  "source": "Test",
  "startIdx": 0,
  "type": "Description",
}
`;

exports[`descriptionTokenizer should tokenize to the next whitespace 1`] = `
Object {
  "endIdx": 4,
  "source": "Test",
  "startIdx": 0,
  "type": "Description",
}
`;
